dns/draft-ct-over-dns.md

```diff
@@ -0,0 +1,289 @@
+---
+title: Certificate Transparency over DNS
+docname: draft-ct-over-dns-01-dev
+category: exp
+pi: [toc, sortrefs, symrefs]
+ipr: trust200902
+area: Security
+wg: TRANS
+kw: Internet-Draft
+
+author:
+  -
+    ins: B. Laurie
+    name: Ben Laurie
+    email: benl@google.com
+    org: Google UK Ltd.
+  -
+    ins: P. Phaneuf
+    name: Pierre Phaneuf
+    email: pphaneuf@google.com
+    org: Google UK Ltd.
+  -
+    ins: A. Eijdenberg
+    name: Adam Eijdenberg
+    email: eijdenberg@google.com
+    org: Google Inc.
+
+normative:
+  RFC6962:
+  RFC1035:
+  RFC4648:
+
+informative:
+  RFC1034:
+
+--- abstract
+
+We describe a method for fetching certificate transparency inclusion proofs over recursive DNS.
+
+--- middle
+
+# Introduction
+
+Certificate Transparency improves the certificate ecosystem by requiring certificates be submitted to logs in order to be considered secure.  To prove that a certificate has been included in a public log clients should request inclusion proofs from logs as descrbed in RFC6962.
+
+However by fetching an inclusion proof directly from a log server the client reveals to the log server the certificate that they are interested in, and thus leaks their browsing history.
+
+This document describes the use of special DNS records so that a client can look up certificate transparency information via their existing DNS resolver, meaning that they will identify their browsing history to only their existing DNS resolver (typically at their ISP), which, in most cases, already knows which sites the user has visited by virtue of having resolved their DNS requests already.
+
+In this manner only a proxy for aggregate usage of sites contained in a log will be detectable by the log operator or mirror that operates the authoritative name server used by these resolvers.
+
+Google plans to operate such a DNS resolver to put in front of log mirrors for at least all logs included by Chrome.  Other clients are welcome to use the same name servers if desired.
+
+# Overview
+
+...
+
+# Messages
+A log operator or mirror conforming with this specification SHALL provide a name server that provides authoritative answers to the following types of queries.
+
+## STH Query
+
+TODO
+
+## Hash Query
+
+Inputs:
+: 
+: `domain_for_log` is a value stored in log metadata.  For example, a name server operated by Google for the Pilot log may have the domain `pilot.ct.googleapis.com`.
+
+: `leaf_hash` is the a leaf hash as defined in section 3.4 of RFC6962.
+
+Outputs:
+: 
+: `leaf_index` is the index of the `leaf_hash` in the Merkle tree.
+
+
+To query for this message, set `QNAME` to `<encoded_leaf_hash> || '.hash.' || <domain_for_log> || '.'`
```

Comment:
You'll need to specify somewhere that `||` denote concatenation.

Or perhaps just say `<encoded leaf hash>.hash.<domain for log>.`